drivers/spi/spi_max32.c

```diff
@@ -565,6 +645,99 @@ static int transceive_dma(const struct device *dev, const struct spi_config *con
 }
 #endif /* CONFIG_SPI_MAX32_DMA */
 
+#ifdef CONFIG_SPI_RTIO
+static void spi_max32_iodev_complete(const struct device *dev, int status);
+
+static void spi_max32_iodev_start(const struct device *dev)
+{
+	struct max32_spi_data *data = dev->data;
+	struct spi_rtio *rtio_ctx = data->rtio_ctx;
+	struct rtio_sqe *sqe = &rtio_ctx->txn_curr->sqe;
+	int ret = 0;
+
+	switch (sqe->op) {
+	case RTIO_OP_RX:
+	case RTIO_OP_TX:
+	case RTIO_OP_TINY_TX:
+	case RTIO_OP_TXRX:
+		ret = spi_max32_transceive(dev);
+		break;
+	default:
+		spi_max32_iodev_complete(dev, -EINVAL);
+		break;
+	}
+	if (ret == 0) {
+		spi_max32_iodev_complete(dev, 0);
+	}
```

Comment:
Shouldn't this be handled asynchronously? Otherwise this is a blocking context

Reply:
Good catch

Reply:
Hi @ubieda,
I've reviewed what you said here and took another look at spi_mcux_lpspi. I've modified code to do the async part using Interrupts can you please take a look and see if this is ok?

Reply:
Is the non-interrupt case working synchronously? Shouldn't we limit RTIO to work only with the asynchronous approach?